server/meshmodel/helper.go

```diff
@@ -0,0 +1,135 @@
+	regManager       *meshmodel.RegistryManager
+	componentChan    chan v1alpha1.ComponentDefinition
+	relationshipChan chan v1alpha1.RelationshipDefinition
+	doneSignal       chan bool
+	errorChan        chan error
+}
+
+func (ch *ComponentHelper) Init(hc *models.HandlerConfig, rm *meshmodel.RegistryManager) {
+	ch.handlerConfig = hc
+	ch.regManager = rm
+	ch.componentChan = make(chan v1alpha1.ComponentDefinition, 1)
+	ch.relationshipChan = make(chan v1alpha1.RelationshipDefinition, 1)
+	ch.doneSignal = make(chan bool)
+	ch.errorChan = make(chan error)
+}
+
+// seed the local meshmodel components
+func (ch ComponentHelper) SeedComponents() {
+	// Read component and relationship definitions from files and send them to channels
+	ch.generateComponents("/components")
+	ch.generateRelationships("/relationships")
+
+	// Register components and relationships with the registry manager
+	go ch.watchComponents()
+}
+
+func (ch ComponentHelper) generateComponents(pathToComponents string) {
+	path, err := filepath.Abs(pathToComponents)
+	if err != nil {
+		ch.errorChan <- errors.Wrapf(err, "error while getting absolute path for generating components")
+		return
+	}
+
+	err = filepath.Walk(path, func(path string, info fs.FileInfo, err error) error {
+		if info == nil {
+			return nil
+		}
+
+		if !info.IsDir() {
+			var comp v1alpha1.ComponentDefinition
+			byt, err := os.ReadFile(path)
+			if err != nil {
+				return err
+			}
+			err = json.Unmarshal(byt, &comp)
+			if err != nil {
+				return err
+			}
+			if comp.Metadata != nil && comp.Metadata["published"] == true {
+				utils.WriteSVGsOnFileSystem(&comp)
+				ch.componentChan <- comp
+			}
+		}
+		return nil
+	})
+	ch.errorChan <- errors.Wrapf(err, "error while generating components")
+	return
+}
+
+func (ch ComponentHelper) generateRelationships(pathToComponents string) {
+	path, err := filepath.Abs(pathToComponents)
+	if err != nil {
+		ch.errorChan <- errors.Wrapf(err, "error while getting absolute path for generating relationships")
+		return
+	}
+
+	err = filepath.Walk(path, func(path string, info fs.FileInfo, err error) error {
+		if info == nil {
+			return nil
+		}
+		if !info.IsDir() {
+			var rel v1alpha1.RelationshipDefinition
+			byt, err := os.ReadFile(path)
+			if err != nil {
+				return nil
+			}
+			err = json.Unmarshal(byt, &rel)
+			if err != nil {
+				return nil
+			}
+			ch.relationshipChan <- rel
+		}
+		return nil
+	})
+
+	ch.errorChan <- errors.Wrapf(err, "error while generating relationships")
+	ch.doneSignal <- true
+	return
+}
+
+func (ch ComponentHelper) watchComponents() {
+	var err error
+	for {
+		select {
+		case comp := <-ch.componentChan:
+			err = ch.regManager.RegisterEntity(meshmodel.Host{
+				Hostname: ArtifactHubComponentsHandler,
+			}, comp)
+		case rel := <-ch.relationshipChan:
+			err = ch.regManager.RegisterEntity(meshmodel.Host{
+				Hostname: ArtifactHubComponentsHandler,
+			}, rel)
+		case mhErr := <-ch.errorChan:
+			fmt.Println("err: ", mhErr.Error())
```

Comment:
Thanks for your contribution, please make sure the same logic is in case `mhErr`.

Reply:
I didn't get it

Reply:
Should we set the error to the variable the code can handle at the end?

Reply:
This is watching all the errors while generating components and relationships. If I understand you correctly, you're suggesting handling this error like returning it to calling functions, but that is not the purpose of it. It'll run asynchronously, watch errors and print them to console.

Reply:
Ok, I see. You can ignore my suggestion.

Reply:
Please add a code comment to mention this in your next commit, thanks.